test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_optimized_sm_training.py

```diff
@@ -0,0 +1,194 @@
+'''
+This testplan will currently testing XLA effectiveness for DLC TF containers.
+'''
+import os
+import re
+import time
+
+from random import Random
+
+import pytest
+
+from invoke.context import Context
+
+from test.test_utils import (
+    BENCHMARK_RESULTS_S3_BUCKET, LOGGER, get_framework_and_version_from_tag, get_cuda_version_from_tag,
+)
+
+
+
+@pytest.mark.flaky(reruns=3)
+@pytest.mark.integration("imagenet dataset")
+@pytest.mark.multinode(4)
+@pytest.mark.model("resnet50")
+def test_optimized_tensorflow_sagemaker_training_performance_multinode(tensorflow_training, region, gpu_only, tf25_and_above_only):
+    throughput_without_xla = run_sm_perf_test(
+                                image_uri=tensorflow_training,
+                                xla=False,
+                                num_nodes=4,
+                                region=region,
+                                threshold=None
+                                )
+    throughput_with_xla = run_sm_perf_test(
+                                image_uri=tensorflow_training,
+                                xla=True,
+                                num_nodes=4,
+                                region=region,
+                                threshold=None
+                                )
+   
+    _, framework_version = get_framework_and_version_from_tag(tensorflow_training)
+    py_version = "py2" if "py2" in tensorflow_training else "py37" if "py37" in tensorflow_training else "py3"
+    '''
+    This is a data collection effort for now.
+    There is enough variation in performance to warrant putting down a THRESHOLD in terms of performance ratio
+    assert throughput_with_xla > throughput_without_xla, (
+        f"tensorflow {framework_version} sagemaker training gpu {py_version} imagenet 4 nodes "
+        f"XLA Benchmark Result {throughput_with_xla} does not reach the threshold {throughput_without_xla}"
+    )
+    '''
+
+
+@pytest.mark.integration("imagenet dataset")
+@pytest.mark.model("resnet50")
+def test_optimized_tensorflow_sagemaker_training_performance_singlenode(tensorflow_training, region, gpu_only, tf25_and_above_only):
+    throughput_without_xla = run_sm_perf_test(
+                                image_uri=tensorflow_training,
+                                xla=False,
+                                num_nodes=1,
+                                region=region,
+                                threshold=None
```

Comment:
Would this threshold be always `None` or this will be added at a later time?

Reply:
For the run without XLA, we won't be needing an absolute threshold. For the run with XLA, we will add a threshold in the future. I will add a TODO for that.